sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/CertificateValidationCheck.cs

```diff
@@ -0,0 +1,299 @@
+
+        private static bool IsValidationDelegateType(ITypeSymbol type)
+        {
+            if (type.Is(KnownType.System_Net_Security_RemoteCertificateValidationCallback))
+            {
+                return true;
+            }
+            else if (type is INamedTypeSymbol NamedSymbol && type.Is(KnownType.System_Func_T1_T2_T3_T4_TResult))
+            {
+                //HttpClientHandler.ServerCertificateCustomValidationCallback uses Func<HttpRequestMessage, X509Certificate2, X509Chain, SslPolicyErrors, bool>
+                //We're actually looking for Func<Any Sender, X509Certificate2, X509Chain, SslPolicyErrors, bool>
+                var parameters = NamedSymbol.DelegateInvokeMethod.Parameters;
+                return parameters.Length == 4   //And it should! T1, T2, T3, T4
+                    && parameters[0].Type.IsClassOrStruct() //We don't care about common (Object) nor specific (HttpRequestMessage) type of Sender
+                    && parameters[1].Type.Is(KnownType.System_Security_Cryptography_X509Certificates_X509Certificate2)
+                    && parameters[2].Type.Is(KnownType.System_Security_Cryptography_X509Certificates_X509Chain)
+                    && parameters[3].Type.Is(KnownType.System_Net_Security_SslPolicyErrors)
+                    && NamedSymbol.DelegateInvokeMethod.ReturnType.Is(KnownType.System_Boolean);
+            }
+            return false;
+        }
+
+        private static ImmutableArray<Location> ArgumentLocations(InspectionContext c, ExpressionSyntax expression)
+        {
+            var ret = ImmutableArray.CreateBuilder<Location>();
+            switch (expression)
+            {
+                case IdentifierNameSyntax identifier:
+                    var identSymbol = c.Context.SemanticModel.GetSymbolInfo(identifier).Symbol;
+                    if (identSymbol != null && identSymbol.DeclaringSyntaxReferences.Length == 1)
+                    {
+                        ret.AddRange(IdentifierLocations(c, identSymbol.DeclaringSyntaxReferences.Single().GetSyntax()));
+                    }
+                    break;
+                case ParenthesizedLambdaExpressionSyntax lambda:
+                    if ((lambda.Body as LiteralExpressionSyntax)?.Kind() == SyntaxKind.TrueLiteralExpression)
+                    {
+                        ret.Add(lambda.Body.GetLocation());   //Code was found guilty for lambda (...) => true
+                    }
+                    else if (lambda.Body is BlockSyntax block)
+                    {
+                        ret.AddRange(BlockLocations(block));
+                    }
+                    break;
+                case InvocationExpressionSyntax invocation:
+                    var invSymbol = c.Context.SemanticModel.GetSymbolInfo(invocation).Symbol;
+                    if (invSymbol != null && invSymbol.DeclaringSyntaxReferences.Length == 1 && invSymbol.DeclaringSyntaxReferences.Single().GetSyntax() is MethodDeclarationSyntax syntax)
```

Comment:
_subjective_ same here, I find it more expressive to avoid local vars and null-checks if not necessary

```
if (c.Context.SemanticModel.GetSymbolInfo(invocation).Symbol is ISymbol invSymbol && ...)

```

Reply:
Same as [above](https://github.com/SonarSource/sonar-dotnet/pull/2785#discussion_r346245941)